performance_test/CMakeLists.txt

```diff
@@ -29,10 +23,23 @@ set(OPTIONAL_LIBRARIES)
 
 # Default to C++14
 set(CMAKE_CXX_STANDARD 14)
-find_package(fastrtps_cmake_module REQUIRED)
 find_package(Boost COMPONENTS program_options REQUIRED)
 find_package(ament_cmake REQUIRED)
-find_package(FastRTPS REQUIRED)
+
+# FastRTPS
+find_package(fastrtps_cmake_module QUIET)
+find_package(FastRTPS QUIET)
+if(FastRTPS_FOUND)
```

Comment:
Sorry, I found one more thing here. What happens if the user sets `PERFORMANCE_TEST_FASTRTPS_ENABLED` but it cannot be found? I guess then compiling fails? Maybe it is better to write:

```
if(PERFORMANCE_TEST_FASTRTPS_ENABLED)
  find_package(fastrtps_cmake_module QUIET REQUIRED)
  find_package(FastRTPS QUIET REQUIRED)
  add_definitions(-DPERFORMANCE_TEST_FASTRTPS_ENABLED)
endif()
```
I guss this is then also the simpler logic.

Reply:
If it's not found, then `add_definitions(-DPERFORMANCE_TEST_FASTRTPS_ENABLED)` is not called, so the intention was that compilation shouldn't fail. But I just noticed that `fast_rtps_communicator.hpp` is not properly guarded, I'll fix that.

With the logic you propose, FastRTPS will be disabled by default. Do you think that's ok?

(As a side-note, I had considered using two separate variables, a user-facing one and another for internal use, but decided that using the same one everywhere would make the code clearer and more maintainable.)

Reply:
See https://github.com/ApexAI/performance_test/pull/20/commits/d92e91e48c23a7128e0d20dcdfacdae4fcc9def0 which should address your point above

Reply:
What happens if I say I set `PERFORMANCE_TEST_FASTRTPS_ENABLED` but it is not found? Would it now not still just compile fine and the user would be confused when trying to run the application? My code would fail if someone says  `PERFORMANCE_TEST_FASTRTPS_ENABLED` and it is not found and due to `REQUIRED` it would give proper information about the error. Or what is your desired behavior for this case?

Reply:
Yeah you're right, I was overcomplicating things. Check out 4bce773, I used your logic and made it enabled by default.